client/src/app/site/base/components/base.component.ts

```diff
@@ -123,7 +125,19 @@ export abstract class BaseComponent implements OnDestroy {
             this.messageSnackBar.dismiss();
         }
 
-        this.cleanSubjects();
+        this.cleanSubscriptions();
+    }
+
+    public updateSubscription(name: string, subscription: Subscription): void {
```

Comment:
This is a solution to an issue I often have: Subscribing to an observable based on a changing fact. E.g. subscribing to an assignment by id from the url -> you have the route observable and inside it's `subscribe` you have to (a) unsubscribe the old assignment subscription (if any) and (b) subscribe to the new assignment. Also you have to clean up on `ngOnDestroy`. This function allows for repeaded one-active-only subscriptions, so doing

```
this.updateSubscription("assignment", <some getter>.subscribe(<some code>));
this.updateSubscription("assignment", <some getter>.subscribe(<some code>));
this.updateSubscription("assignment", <some getter>.subscribe(<some code>));
```

is harmless since it cleans up when subscribing to a new subscription and it is cleaned up in `ngOnDestroy`. So we now have two systems for subscriptions: You can push into `this.subscriptions` (e.g. useful in the CTOR, or when many things are registered just once` and the new one for repeaded-subscriptions.

Reply:
Having a route subscription is nice when you never leave your component, i.e going back and forth in motions or only navigating between assignments using the full screen search components.
Otherwise, if you need it synchronously, you could also access the activated route directly and extract the router patameters